### PR TITLE
Bumping to 1.4.1+

### PR DESCRIPTION
**Pull request by jmihaliczas3d**
## Jira ticket URL (Why?)
[BUILD-1544](https://shapr3d.atlassian.net/browse/BUILD-1544)

## Short description (What? How?) 📖
Merging upstream/`v1.x` at `5ebfc927306fd7ce551fa22244be801cf2b9fdd9`, which contains new commits, including fixes, on top of the `1.14.1` release.
See https://github.com/shapr3d/shapr3d/pull/20784 as the main point of discussion about the change this is a part of.

If approved, I'll merge this manually to avoid squash.

### Design review 🎨
n/a

### Affected areas 🧭
all

### How did you test it? 🤔
Manual 💁‍♂️

Automated 💻
Successful CI runs:
* [Apple Unit](https://github.com/shapr3d/shapr3d/actions/runs/10559700331)
* [Apple UI](https://github.com/shapr3d/shapr3d/actions/runs/10559695699)
* [Apple Render](https://github.com/shapr3d/shapr3d/actions/runs/10559703481)
* [Windows Unit](https://github.com/shapr3d/shapr3d/actions/runs/10561324714)
* [Windows UI](https://github.com/shapr3d/shapr3d/actions/runs/10561328348)
* [Windows Render](https://github.com/shapr3d/shapr3d/actions/runs/10561332301)

[BUILD-1544]: https://shapr3d.atlassian.net/browse/BUILD-1544?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lowdesertpunk** [comment]
> If approved, I'll merge this manually to avoid squash.

Enabled merge commits for this repo, so no need to merge manually (I mean not via GitHub's web interface).

**Comment by lowdesertpunk** [comment]
Did you actually review commits since the [last reviewed one](https://github.com/shapr3d/spdlog/pull/4/commits/1dd942cf060239676428042a2cfbfc5dd2545cbe)? Last time we also found actual platform (iOS) related runtime issues.

**Comment by lowdesertpunk** [comment]
I would rather merge the first PR first and then rebase this one so we can do a proper review.

**Comment by lowdesertpunk** [comment]
Merged, please rebase.

File: include/spdlog/details/registry.h [hunk 1]
```diff
@@ -42,8 +42,10 @@ class SPDLOG_API registry {
     // another.
     logger *get_default_raw();
 
-    // set default logger.
+    // set default logger and add it to the registry if not registered already.
     // default logger is stored in default_logger_ (for faster retrieval) and in the loggers_ map.
+    // Note: Make sure to unregister it when no longer needed or before calling again with a new
+    // logger.
```
Comment (jmihaliczas3d) [inline include/spdlog/details/registry.h:48]: This is a change in the contract, unregister becomes the client's responsibility

File: include/spdlog/details/registry.h [hunk 2]
```diff
@@ -68,6 +70,11 @@ class SPDLOG_API registry {
         periodic_flusher_ = details::make_unique<periodic_worker>(clbk, interval);
     }
 
+    std::unique_ptr<periodic_worker> &get_flusher() {
+        std::lock_guard<std::mutex> lock(flusher_mutex_);
+        return periodic_flusher_;
+    }
```
Comment (jmihaliczas3d) [inline include/spdlog/details/registry.h:76]: I am not sure this makes much sense here. This local lock object will not limit concurrent access to the object via the returned reference.

**Comment by jmihaliczas3d** [comment]
I reviewed the spdlog commits, but not the fmt major upgrade.

There was a fix for fmt 11.0.0 compatibility, which means someone tried or uses that combo.

**Comment by jmihaliczas3d** [comment]
> Did you actually review commits since the [last reviewed one](https://github.com/shapr3d/spdlog/pull/4/commits/1dd942cf060239676428042a2cfbfc5dd2545cbe)? Last time we also found actual platform (iOS) related runtime issues.

Do you think we should check the `fmt` commits one by one too?

**Comment by palver123** [comment]
Is there anyone else who has some experience with fmt or spdlog? If yes, they would be better reviewers. If nobody, then I'll dig into it, as I currently have no idea how to prove that this bump is OK or not.

**Comment by lowdesertpunk** [comment]
> Is there anyone else who has some experience with fmt or spdlog? If yes, they would be better reviewers. If nobody, then I'll dig into it, as I currently have no idea how to prove that this bump is OK or not.

No need. Don't even know why you were added as a reviewer :) Feel free to remove yourself.

File: include/spdlog/details/registry.h (hunk 1 above)
Comment (lowdesertpunk) [reply include/spdlog/details/registry.h:48]: Only applies to the default logger, which we only set once, so it's fine.

File: include/spdlog/details/registry.h (hunk 2 above)
Comment (lowdesertpunk) [reply include/spdlog/details/registry.h:76]: But it does protect the actual instance used by the registry. Anyway we are not using it, so fine.

**Review by lowdesertpunk — APPROVED** [review]

**Comment by lowdesertpunk** [comment]
> Do you think we should check the `fmt` commits one by one too?

I've only looked for possible platform specifics, but none found. I would say this should suit.

**Pull request merged**